app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb

```diff
@@ -191,6 +191,14 @@ def vms_by_id
     @vms_by_id ||= Hash[vms.collect { |s| [s.id, s] }]
   end
 
+  def cloud_volumes
+    return [] if references(:cloud_volumes).blank?
+    return @cloud_volumes if @cloud_volumes.any?
+    @cloud_volumes = references(:cloud_volumes).collect do |volume_id|
+      safe_get { cinder_service.volumes.get(volume_id) }
```

Comment:
@mansam just a sanity check, ```cinder_service.volumes.get(volume_id) ``` works across differnet tenants, right?

Reply:
It doesn't, sadly. `cinder_service.volumes.all(:id => volume_id, :all_tenants => true)` works, but that requires admin. I might have to just make a scoped request for each volume.

Reply:
right, so probably lets target this as perf issues for next minor release? Could you do some perf test with 200-500 tenants (and more)? That is what we see in production. Then lets talk to Loic where to target this.

Or maybe I understand it badly? How many API queries will this cause with having 500 tenants?